pingrid/impl.py

```diff
@@ -402,14 +610,174 @@ def produce_shape_tile(
+BLACK = Color(0, 0, 0)
+BLUE = Color(0, 0, 255)
+BROWN = Color(165, 42, 42)
+DARKORANGE = Color(255, 140, 0)
+DARKRED = Color(128, 0, 0)
+DEEPSKYBLUE = Color(0, 191, 255)
+LIMEGREEN = Color(50, 205, 50)
+MOCCASIN = Color(255, 228, 181)
+NAVY = Color(0, 0, 128)
+ORANGE = Color(255, 165, 0)
+PALEGREEN = Color(152, 251, 152)
+PURPLE = Color(160, 32, 240)
+RED = Color(255, 0, 0)
+TURQUOISE = Color(64, 224, 208)
+WHITE = Color(255, 255, 255)
+YELLOW = Color(255, 255, 0)
+
+_CORRELATION_CS = ColorScale(
+    "correlation",
+    [NAVY, BLUE, DEEPSKYBLUE, AQUAMARINE, PALEGREEN, MOCCASIN, MOCCASIN, YELLOW, DARKORANGE, RED, DARKRED],
+    [-1, -0.8, -0.6, -0.3, -0.1, -0.1, 0.1, 0.1, 0.4, 0.7, 1],
+)
+
+_RAINBOW_CS = ColorScale("rainbow", [
+    Color(0, 0, 255),
+    Color(0, 255, 255),
+    Color(0, 255, 0),
+    Color(255, 255, 0),
+    Color(255, 0, 0),
+    Color(255, 0, 255),
+])
+
+_PRECIP_CS = ColorScale(
+    "precip",
+    [
+        WHITE,
+        WHITE,
+        Color(210, 255, 215),
+        Color(210, 255, 215),
+        Color(150, 230, 155),
+        Color(150, 230, 155),
+        Color(110, 210, 115),
+        Color(110, 210, 115),
+        Color(45, 180, 50),
+        Color(45, 180, 50),
+        Color(20, 170, 25),
+        Color(20, 170, 25),
+        Color(10, 150, 15),
+        Color(10, 150, 15),
+        Color(0, 130, 5),
+        Color(0, 130, 5),
+        Color(0, 110, 4),
+        Color(0, 110, 4),
+    ],
+    [0, 0.2, 0.2, 2, 2, 4, 4, 6, 6, 8, 8, 10, 10, 12, 12, 14, 14, 16],
+)
+
+_RAIN_POE_CS = ColorScale(
+    "rain_poe",
+    [BLACK, BROWN, ORANGE, YELLOW, MOCCASIN, MOCCASIN, LIMEGREEN, TURQUOISE, BLUE, PURPLE],
+    [0, 0.15, 0.30, 0.45, 0.45, 0.55, 0.55, 0.7, 0.85, 1],
+)
+
+_RAIN_PNE_CS = _RAIN_POE_CS.reversed(name="rain_pne")
+
+_VULN_CS = ColorScale(
+    "vulnerability",
+    [
+        YELLOW,
+        YELLOW,
+        Color(255, 245, 0),
+        Color(255, 245, 0),
+        Color(255, 222, 0),
+        Color(255, 222, 0),
+        Color(255, 195, 0),
+        Color(255, 195, 0),
+        Color(255, 174, 0),
+        Color(255, 174, 0),
+        Color(255, 153, 0),
+        Color(255, 153, 0),
+        Color(255, 126, 0),
+        Color(255, 126, 0),
+        Color(255, 98, 0),
+        Color(255, 98, 0),
+        Color(255, 70, 0),
+        Color(255, 70, 0),
+        Color(255, 47, 0),
+        Color(255, 47, 0),
+        Color(255, 11, 0),
+        Color(255, 11, 0),
+        Color(247, 0, 0),
+        Color(247, 0, 0),
+        Color(219, 0, 0),
+        Color(219, 0, 0),
+        Color(191, 0, 0),
+        Color(191, 0, 0),
+        Color(166, 0, 0),
+        Color(166, 0, 0),
+        Color(138, 0, 0),
+        Color(138, 0, 0),
+        Color(128, 0, 0),
+        Color(128, 0, 0),
+    ],
+    [0, 1, 1, 1.2, 1.2, 1.4, 1.4, 1.6, 1.6, 1.7, 1.7, 1.8, 1.8,
+     2, 2, 2.2, 2.2, 2.4, 2.4, 2.6, 2.6, 2.8, 2.8,
+     3, 3, 3.2, 3.2, 3.4, 3.4, 3.6, 3.6, 3.8, 3.8, 5],
+)
+
+_FBF_PNE_CS = ColorScale(
+    "fbf_pnep",
```

Comment:
We shouldn't have an item called `fbf_pnep` here. Either it's reusable in other applications, in which case it should be called something more generic like `pnep`,  or it's specific to fbf, in which case it should be in the fbf application, not in pingrid.